bndtools.m2e/src/bndtools/m2e/IndexConfigurator.java

```diff
@@ -176,12 +178,31 @@ public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
 				}
 
 				if (needsBuild) {
-					SubMonitor buildMonitor = SubMonitor.convert(monitor,
-						"Rebuilding index for project " + facade.getProject()
+
+					SubMonitor indexMonitor = SubMonitor.convert(monitor,
+						"Rebuilding indexes for project " + facade.getProject()
+							.getName(),
+						2);
+					List<MojoExecution> list = facade.getMojoExecutions(INDEXER_PLUGIN_GROUP_ID,
+						INDEXER_PLUGIN_ARTIFACT_ID, indexMonitor);
+
+					if (list.isEmpty()) {
+						indexMonitor.done();
+					} else {
+						SubMonitor buildMonitor = SubMonitor.convert(monitor,
+							"Rebuilding indexes for project " + facade.getProject()
 							.getName(),
-						1);
-					facade.getProject()
-						.build(FULL_BUILD, buildMonitor);
+							list.size());
+						list.forEach(me -> {
+							try {
+								MavenPlugin.getMaven()
+									.execute(getMavenProject(facade, indexMonitor), me, indexMonitor);
+							} catch (CoreException e) {
+								// TODO Auto-generated catch block
+								e.printStackTrace();
```

Comment:
This should be logged to the eclipse logger.

Reply:
Oops - fixed, amended and force pushed